src/main/java/appeng/me/cache/CraftingGridCache.java

```diff
@@ -416,7 +416,7 @@ public ImmutableCollection<ICraftingPatternDetails> getCraftingFor(
             final World world) {
         final ImmutableList<ICraftingPatternDetails> res;
         boolean normalMode = false;
-        if (details != null && details.canSubstitute()) {
+        if (details != null && details.canBeSubstitute()) {
```

Comment:
This is not resolved yet

Reply:
This should probably not be changed. `details` is the pattern in which `whatToCraft` is an ingredient instead of output. It's correct to check for `canSubstitute()`